src/tracer.cc

```diff
@@ -440,6 +641,17 @@ TraceManager::TraceActivity(
       reinterpret_cast<std::shared_ptr<TraceManager::Trace>*>(userp)->get();
 
   std::lock_guard<std::mutex> lk(ts->mtx_);
+
+  if (ts->setting_->mode_ == TRACE_MODE_OPENTELEMETRY) {
+#ifndef _WIN32
+    ts->ReportToOpenTelemetry(trace, activity, timestamp_ns);
+#else
+    LOG_ERROR << "Unsupported trace mode: "
+              << TraceManager::InferenceTraceModeString(mode_);
+#endif
+    return;
+  }
```

Comment:
For future improvement, we can derive subclass from `Trace` for different mode, i.e. `OpenTelemetryTrace`, `TritonTrace` and they implements a `Report()` function which is going to be `ReportToOpenTelemetry` or the `stringstream` appending below

Reply:
Thanks for this idea! I will add this to the refactoring ticket